### ISSUE TITLE
cabal init -n sometimes puts two newlines instead of one in .cabal file

### ISSUE THREAD

**Issue by fgaz** [haskell/cabal#8236]
**Describe the bug**

cabal init terminates the .cabal file with two newlines instead of one

**To Reproduce**

```
$ cabal init -n
[...]
$ tail *.cabal

    -- Other library packages from which modules are imported.
    build-depends:    base ^>=4.16.1.0

    -- Directories containing source files.
    hs-source-dirs:   app

    -- Base language which the package is written in.
    default-language: Haskell2010

```

**Expected behavior**

Files should be terminated with a single newline character

**System information**
 - cabal HEAD (a715d0473b0bca458a625ab308cef2f65d854799)

cc @ptkato @emilypi

**Comment by emilypi** [comment]
Hmmm. I wonder if `writeFile` or the formatting leaks an additional newline the way we do file creation. @ptkato care to investigate?

**Comment by ulysses4ever** [comment]
While working on a fix (#8292), I realized it's actually a bit worse than initially described:  two new lines in a row can appear not only in the very end, but between sections too, depending on what sections materialized. Here's an example:
```
❯ ~/bin/cabal-head --version
Warning: this is a debug build with assertions enabled.
cabal-install version 3.9
compiled using version 3.9.0.0 of the Cabal library 

❯ ~/bin/cabal-head init -n -m
Warning: this is a debug build with assertions enabled.
[Log] Guessing dependencies...
[Log] Using cabal specification: 3.6
[Warning] unknown license type, you must put a copy in LICENSE yourself.
[Log] Creating fresh file CHANGELOG.md...
[Log] Creating fresh directory ./app...
[Log] Creating fresh file app/Main.hs...
[Log] Creating fresh file A.cabal...
[Warning] No synopsis given. You should edit the .cabal file and add one.
[Info] You may want to edit the .cabal file and add a Description field.


❯ cat A.cabal
cabal-version:   3.6
name:            A
version:         0.1.0.0
license:         NONE
author:          a.pelenitsyn@gmail.com
maintainer:      Artem Pelenitsyn
build-type:      Simple
extra-doc-files: CHANGELOG.md

common warnings
    ghc-options: -Wall


executable A
    import:           warnings
    main-is:          Main.hs
    build-depends:    base ^>=4.15.1.0
    hs-source-dirs:   app
    default-language: Haskell2010


```
Note the two newlines between the common stanza and the executable one. Also, note the extra trailing newline in the log output of `cabal init -n` — this is unrelated, but #8292 fixes it too. 

Please, consider giving #8292 a review, so that it makes it into 3.8.
